app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt

```diff
@@ -413,12 +433,23 @@ class GradesFragment : FragmentForAccessingTumOnline<ExamList>(
 
         fun newInstance() = GradesFragment()
 
+        private val uncommonGradeRe = Regex("[1-5],[1245689][0-9]*")
+        private val longGradeRe = Regex("[1-4],[0-9][0-9]+")
+
         private val GRADE_COLORS = intArrayOf(
-            R.color.grade_1_0, R.color.grade_1_3, R.color.grade_1_4, R.color.grade_1_7,
-            R.color.grade_2_0, R.color.grade_2_3, R.color.grade_2_4, R.color.grade_2_7,
-            R.color.grade_3_0, R.color.grade_3_3, R.color.grade_3_4, R.color.grade_3_7,
-            R.color.grade_4_0, R.color.grade_4_3, R.color.grade_4_4, R.color.grade_4_7,
-            R.color.grade_5_0, R.color.grade_default
+                R.color.grade_1_0, R.color.grade_1_1, R.color.grade_1_2, R.color.grade_1_3,
+                R.color.grade_1_4, R.color.grade_1_5, R.color.grade_1_6, R.color.grade_1_7,
+                R.color.grade_1_8, R.color.grade_1_9,
+                R.color.grade_2_0, R.color.grade_2_1, R.color.grade_2_2, R.color.grade_2_3,
+                R.color.grade_2_4, R.color.grade_2_5, R.color.grade_2_6, R.color.grade_2_7,
+                R.color.grade_2_8, R.color.grade_2_9,
+                R.color.grade_3_0, R.color.grade_3_1, R.color.grade_3_2, R.color.grade_3_3,
+                R.color.grade_3_4, R.color.grade_3_5, R.color.grade_3_6, R.color.grade_3_7,
+                R.color.grade_3_8, R.color.grade_3_9,
+                R.color.grade_4_0, R.color.grade_4_1, R.color.grade_4_2, R.color.grade_4_3,
+                R.color.grade_4_4, R.color.grade_4_5, R.color.grade_4_6, R.color.grade_4_7,
+                R.color.grade_4_8, R.color.grade_4_9,
+                R.color.grade_5_0, R.color.grade_default
```

Comment:
This makes me think, if it just wouldn't be simpler to round the grades to the closest known grade without the additional inbetween steps. But eh, now you already did the work so we can leave it as is.